### PR TITLE
Implement cudaq::control() taking a free function as the callable arg…

### PR DESCRIPTION
**Pull request by schweitzpgi**
…ument.

### CODE REVIEW

**Comment by schweitzpgi** [comment]
No test. The plan is to extend the test added in an earlier PR for `cudaq::control()`.

**Review by amccaskey — APPROVED** [review]

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,133 @@
+stages:
```
Comment (bettinaheim) [inline .gitlab-ci.yml:1]: This file shouldn't be here.

**Comment by bettinaheim** [comment]
@schweitzpgi Could you please link an issue with a brief outline for what support is added? This will also facilitate capturing it in release notes.

**Comment by schweitzpgi** [comment]
> @schweitzpgi Could you please link an issue with a brief outline for what support is added? This will also facilitate capturing it in release notes.

Internal issue 381.

**Comment by bettinaheim** [comment]
Rel notes: see also https://github.com/NVIDIA/cuda-quantum/pull/32.

**Pull request merged**